psycop/projects/AcuteSomaticAdmission/CohortDefinition/get_psychiatric_outpatient_visits.py

```diff
@@ -0,0 +1,62 @@
+"""
+Script for obtaining and writing all ambulant psychiatric visits. Handles LPR2 to LPR3 transition and duplicates
+"""
+from datetime import timedelta
+
+import pandas as pd
+
+from psycop.common.global_utils.sql.loader import sql_load
+from psycop.common.global_utils.sql.writer import write_df_to_sql
+
+def get_outpatient_visits_to_psychiatry(write: bool = False) -> pd.DataFrame:
+    # Load all physical visits data
+    view = "[FOR_besoeg_fysiske_fremmoeder_inkl_2021_feb2022]"
+    cols_to_keep = "datotid_start, datotid_slut, dw_ek_borger, psykambbesoeg AS pt_type"
+
+    sql = "SELECT " + cols_to_keep + " FROM [fct]." + view
+    sql += "WHERE datotid_start > '2012-01-01' AND psykambbesoeg = 1"
+
+    df = pd.DataFrame(sql_load(sql, chunksize=None))  # type: ignore
+
+    df[["datotid_start", "datotid_slut"]] = df[["datotid_start", "datotid_slut"]].apply(
+        pd.to_datetime
+    )
+
+    # Subtract 1 day from datotid_start in ambulant dates because we want to make predictions one day prior to visit
+    # Even if it is the first psychiatric contact we would still like to make a prediction before
+    # the visit because the patient might have information from somatic visits
+    df["datotid_predict"] = df["datotid_start"] - timedelta(days=1)  # type: ignore
+
+    df = df.drop_duplicates(subset=["dw_ek_borger", "datotid_predict"])
+
+    if write:
+        ROWS_PER_CHUNK = 5_000
+
+        write_df_to_sql(
+            df=df[["dw_ek_borger", "datotid_predict"]],  # type: ignore
+            table_name="all_psychiatric_outpatient_visits_processed_2012_2021_ANDDAN_SOMATIC_ADMISSION",
+            if_exists="replace",
+            rows_per_chunk=ROWS_PER_CHUNK,
+        )
+
+    #Så sætter jeg det korrekte navn - de andre kalder nedenstående funktion der gør det samme. Måske for at spare tid når datasættet loades
+    df = df.rename(columns={"datotid_predict": "timestamp"})
+
+    return df[["dw_ek_borger", "timestamp"]]  # type: ignore
```

Comment:
Du kan vist udskifte hele denne funktion med et kald til:
```python
from psycop.common.feature_generation.loaders.raw.load_visits import ambulatory_visits

prediction_times = pl.from_pandas(
            ambulatory_visits(
                timestamps_only=True,
                timestamp_for_output="start",
                n_rows=None,
                return_value_as_visit_length_days=False,
                shak_code=6600,
                shak_sql_operator="=",
            )
        ).with_columns(pl.col("timestamp") - pl.duration(days=1))
```
`ambulatory_visits` loaderen henter alle besøg til ambulatoriet, med mulighed for at begrænse til en bestemt shakkode (6600 for psykiatrien)

Reply:
Hej Lasse
Når jeg forsøger at køre koden siger den at 'p1' ikke er defineret. Jeg kan ikke helt gennemskue hvor den skal defineres. kan du hjælpe mig?

Reply:
Ah, ja, der mangler at importeres polars. Så tilføj
```python
import polars as pl
```

Reply:
Det fungere strålende - tak :-)